eureka-client/src/test/java/com/netflix/discovery/converters/wrappers/CodecWrappersTest.java

```diff
@@ -0,0 +1,61 @@
+package com.netflix.discovery.converters.wrappers;
+
+import junit.framework.Assert;
+import org.junit.Test;
+
+import javax.ws.rs.core.MediaType;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+
+/**
+ * @author David Liu
+ */
+public class CodecWrappersTest {
+
+    private static String testWrapperName = "FOO_WRAPPER";
+
+    @Test
+    public void testRegisterNewWrapper() {
+        Assert.assertNull(CodecWrappers.getEncoder(testWrapperName));
+        Assert.assertNull(CodecWrappers.getDecoder(testWrapperName));
+
+        CodecWrappers.registerWrapper(new TestWrapper());
+
+        Assert.assertNotNull(CodecWrappers.getEncoder(testWrapperName));
+        Assert.assertNotNull(CodecWrappers.getDecoder(testWrapperName));
+    }
+
+    private final class TestWrapper implements EncoderDecoderWrapper {
```

Comment:
The initial purpose of adding this was to aid in the migration of legacy codecs to new codecs, with the ability to switch between codecs quickly (without having rebuild and redeploy) for compatibility purposes. In the case of the Jackson json codecs for example, the two codecs does not actually have any dependency differences (both use jackson), but they handle the encode and decode differently.